### PR TITLE
Creation of release v0.10.0

### LINKED ISSUE #263: Finalisation of release v0.10.0
**Problem description**

v0.10.0 is available as second release candidate. There were no new issues raised towards this release since January 11th. Therefore we might be ready to declare the release as final and move on.

**Expected action**

* Update CHANGELOG.md (with a sum of changes of RC1+2)
* Update README.md
* Change version number in YAML to 0.10.0

**Additional context**

Operators (like DT) have started to implemented v0.10.0 and waiting for the final release.

### PR DESCRIPTION
**Pull request by hdamker**
#### What type of PR is this?

Add one of the following kinds:
* documentation
* subproject management

#### What this PR does / why we need it:

This PR will create the release v0.10.0 of the QualityOnDemand repository:

- Update CHANGELOG.md (with a sum of changes of RC1+2)
- Update README.md
- Change version number in YAML to 0.10.0

After the PR is agreed and merge of this PR one of the codeowner should:

- Create the release `v0.10.0` within GitHub (use the text for v0.10.0 from CHANGELOG.md as the release description)
- Create the branch `release-0.10.0` on the same commit which is tagged as v0.10.0

- Create a PR to change the version of the OAS YAML within main to `v0.11.0-wip`

#### Which issue(s) this PR fixes:

Fixes #263 

#### Special notes for reviewers:

The links which are containing `release-0.10.0` will get only valid when the PR is merged, and the release branch is created.

Please have a special view on the "Please note:" section within the proposed CHANGELOG.md. Maybe some more information from it should be used also in the README.md

#### Changelog input

```
na
```

### CODE REVIEW

**Review by RandyLevensalor — COMMENTED** [review]
Looks good.  Just one suggestion on wording.

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +15,58 @@ Version numbers 0.2.x to 0.7.x were intentionally not used to avoid conflicts wi
...
+  - OpenAPI [YAML spec file](https://github.com/camaraproject/QualityOnDemand/blob/release-0.10.0/code/API_definitions/qod-api.yaml)
+
+### Please note:
+
+- **This release contains significant changes compared to v0.9.0, and the QoD API is not backward compatible**
+  - Within notifications the schema `EventNotification`has been replace by `CloudEvent` in accordance with the updated CAMARA Design Guidelines
+  - If within `device` an IPv6 address is used it must be a single IPv6 address (out of the prefix used by the device)
+- This release includes changes to be compliant with the [Design Guidelines](https://github.com/camaraproject/Commonalities/blob/release-0.2.0/documentation/API-design-guidelines.md#10-security) and other documents in [release v0.2 of CAMARA Commonalities](https://github.com/camaraproject/Commonalities/tree/release-0.2.0)
+- Please note also that this is still a v0.x release and further releases before the first stable major release might introduce again breaking changes (e.g. to comply with changes agreed within Commonalities)
```
[the remark this answers is not in the document]
Comment (hdamker) [reply CHANGELOG.md:33]: I like the proposal. In addition I will now omit now the "Please note:" as this is already the headline of the section.

**Review by RandyLevensalor — APPROVED** [review]
LGTM

**Review by sfnuser — APPROVED** [review]
Thanks @hdamker. Looks good.

**Review by eric-murray — APPROVED** [review]

**Review by jlurien — APPROVED** [review]

**Pull request merged**